src/app/models/workspace.server.model.ts

```diff
@@ -70,6 +75,34 @@ const compileWorkspaceModel = (db: Mongoose): IWorkspaceModel => {
     return this.findOne({ _id: workspaceId, admin: admin }).exec()
   }
 
+  WorkspaceSchema.statics.deleteWorkspace = async function (
+    workspaceId: IWorkspaceSchema['_id'],
+    admin: IUserSchema['_id'],
+    shouldDeleteForms: boolean,
+  ): Promise<number> {
+    const workspaceToDelete = await this.findOne({
+      _id: workspaceId,
+      admin: admin,
+    })
+    const session = await this.startSession()
+
+    session.startTransaction()
+    const deleted = await this.deleteOne({
+      _id: workspaceId,
+      admin: admin,
+    })
+    if (shouldDeleteForms) {
+      await Form.updateMany(
+        { _id: { $in: workspaceToDelete?.formIds } },
+        { $set: { status: FormStatus.Archived } },
+      )
+    }
+    await session.commitTransaction()
+    session.endSession()
```

Comment:
this logic belongs in the service layer above, not in the model layer.

model methods are meant to deal with operations on that model itself, without knowledge of the business logic. services are meant to coordinate changes to the models according to the business logic.

if you check the other usage of transactions in the codebase (`createTwilioCredentials`) you can see that it takes place in the service layer as well.

Reply:
Updated to move the transaction logic to the service layer!